src/moscot/base/problems/compound_problem.py

```diff
@@ -428,117 +503,57 @@ def add_problem(
     ) -> "BaseCompoundProblem[K, B]":
         """Add a subproblem.
 
-        This function adds and prepares a problem, e.g. if it is not included by the initial
-        :class:`~moscot.utils.subset_policy.SubsetPolicy`.
+        .. seealso::
+            - See :doc:`../notebooks/examples/problems/300_adding_and_removing_problems` on how to add subproblems.
 
         Parameters
         ----------
-        %(key)s
-
+        key
+            Key in :attr:`problems` where to add the subproblem.
         problem
-            Instance of a :class:`~moscot.base.problems.OTProblem`.
+            Subproblem to add.
         overwrite
-            If `True` the problem will be reinitialized and prepared even if a problem with `key` exists.
+            Whether ot overwrite an existing subproblem in :attr:`problems`.
+        kwargs
+            Additional keyword arguments.
 
         Returns
         -------
-        The updated compound problem.
+        Self and updates the following fields:
+
+        - :attr:`problems`
         """
         if TYPE_CHECKING:
             assert isinstance(self._problem_manager, ProblemManager)
         self._problem_manager.add_problem(key, problem, overwrite=overwrite, **kwargs)
         return self
 
-    @d.dedent
     @require_prepare
     def remove_problem(self, key: Tuple[K, K]) -> "BaseCompoundProblem[K, B]":
         """Remove a subproblem.
 
+        .. seealso::
+            - See :doc:`../notebooks/examples/problems/300_adding_and_removing_problems` on how to remove subproblems.
+
         Parameters
         ----------
-        %(key)s
+        key
+            Key of the subproblem to remove.
 
         Returns
         -------
-        The updated compound problem.
+        Self and updates the following fields:
+
+        - :attr:`problems`
         """
         if TYPE_CHECKING:
             assert isinstance(self._problem_manager, ProblemManager)
         self._problem_manager.remove_problem(key)
         return self
 
-    # TODO(MUCKD): should be on the OT problem level as well
-    def save(
```

Comment:
@michalk8 why do we remove this?

Reply:
This is now on the `BaseProblem` level.

Reply:
So we that we can also save the `OTProblem`.